lib/services/semantic_search/frameworks/ml_framework.dart

```diff
@@ -120,6 +171,15 @@ abstract class MLFramework {
     }
   }
 
+  Future<bool> _canDownload() async {
+    final connectivityResult = await (Connectivity().checkConnectivity());
+    bool canDownloadUnderCurrentNetworkConditions = true;
+    if (connectivityResult == ConnectivityResult.mobile) {
```

Comment:
Can be a single line, `connectivityResult != ConnectivityResult.mobile || shouldDownloadOverMobileData`